### DISCUSSION TITLE
wildcards gabage over a few uses

### DISCUSSION THREAD

**Discussion by lgwjames** [lllyasviel/Fooocus#3390] [Q&A]
after using my wildcards to make say 12 images

next load says empty file

open up the txt and its 

˚§d§ƒ¢ñ™*!Œ(¥¢π%Z#P£#^&e©ßÇ†;°>ùû•Ñ"ù#*ùé#G¶x,°ÏSú€#ƒ÷™é$§%0§0"+>√({()íü.€®≠%£˘•‰ù‹ Æ©rß®⁄ï∫®$%$"õ1´qÄ 8©ú)≈&K•Àü«¶<#:!Dèr¢ﬁ&ÚÀ†»P®D™K(}([®‡ßû#ö®[®Zô$†8*,Ñ£
(dù©à!§&0õ%®≥ó›°ë®Y)n*Ñ©
û'Äß +5)∏€ï.,‘j§¢'∂òÍ°:¨[·®º¶<Ì¢b!•$—(©P´±_úòZå¨F"%œ"¿õƒ£†§Ω¶Q´z° ·"K¶ù™#0ùZ©∂$Ñ)ã†R¢a ¶T'›$∞Ö©ú•~®Ö¶,¢ﬁ®õüá®.)ò®¢,®∂ûj"¿é'D¶π¶ï-ë®ö§öÙ©Úûü∑ô˛*¸°d®√$,•ú&r•§üw®£&‚ ﬁG©
†„ó∆•üö@%H©2®”&H%)ö†™5©º)Ïûl$h"¢•÷¶xö˙ü+R*_®&»(äz+Ü¨¢\®r¢Î"N´@®^4$û &'§®[†+&&T°·©8x¶í&m(r@)¿ü"©¬%F&~ß”¶¢®[÷î¨Ë§©ˆüı˝§§új&!$d!•ÍêS "0")4®•$û°≠ùÆï€'≤™õAï´Êü1™N®ﬁ¶F‹£Œ$=ú‚*ú÷¶B¢&äßJ´h' •v£¥§`!™¯$I§˚($~<ú˜´eúö.-7(\†°¢"‹§¢R(®®S$]®k(PöR&¨„&'p,`¢ê$Ã$º'ò¶˜¿ßl(&ß¿©^#÷é∆*§(ú¶Åü…å´Ë®ı!ﬁ)¸•)èñ$l#ˆ§O(¨•Œ(„'“•Ü}¨(•@´`®Ïüòº¢ﬁ°P•!•Û$ö™é|¢Fû`ü¨§%!%â •ü§î£ø§œ†^%B¢<&&J®Níkñ©8™ë!Ñ&@¶JßP

ideals? not just one, most of them

**Comment by lgwjames** [comment]
```
[Fooocus] Processing prompts ...
[Wildcards] processing: __jessicaL__ , { __perspective__ | __emotion__ | __pose__ }
[Wildcards] A English rose woman, blonde hair, blue eyes, freckles around nose and cheeks. 32dd breasts, perfect eyes, perfect hands , { __perspective__ | __emotion__ | __pose__ }
[Wildcards] A English rose woman, blonde hair, blue eyes, freckles around nose and cheeks. 32dd breasts, perfect eyes, perfect hands , { Panoramic | __emotion__ | __pose__ }
[Wildcards] Warning: emotion.txt missing or empty. Using "emotion" as a normal word.
[Wildcards] A English rose woman, blonde hair, blue eyes, freckles around nose and cheeks. 32dd breasts, perfect eyes, perfect hands , { Panoramic | emotion | __pose__ }
[Wildcards] Warning: pose.txt missing or empty. Using "pose" as a normal word.
[Wildcards] A English rose woman, blonde hair, blue eyes, freckles around nose and cheeks. 32dd breasts, perfect eyes, perfect hands , { Panoramic | emotion | pose }
[Wildcards] processing: __neg__
```

emotion has text

amused
cheeky
classy
confident
embarrassed
enticing
excited
extroverted
flamboyant
happy
introverted
nervous
outgoing
proud

pose has text

djusting her collar
arching her back
arm at her side
arms at her side
arms behind her back
arms behind her head
arms crossed
arms up
beckoning
bent over
body bridge
bunny pose
butterfly sitting
chin in hand
claw pose

did i put in the wrong prompt?

**Comment by mashb1t** [comment]
Fooocus does not write anything to wildcards, so it must have been another editor. Please ensure you do save the wildcards files as utf-8 and not unsupported encodings.

What's the reason why you write your prompt as
`__jessicaL__ , { __perspective__ | __emotion__ | __pose__ }`
and not
`__jessicaL__, __perspective__, __emotion__, __pose__`?

**Comment by lgwjames** [comment]
thanks, it's on a Mac and downloaded 300 from the civitai and just put them all in the folder. opened them up and see words are all ok

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by mashb1t** [reply]
> see words are all ok

is in contrast to

> not just one, most of them

so what is it?

**Comment by lgwjames** [comment]
to pick a random thing from each one? is the syntax prompt incorrect?

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by mashb1t** [reply]
the syntax you wrote doesn't exist in Fooocus. To select something randomly you can create a wildcard with 3 lines
```
__perspective__
__emotion__
__pose__
```
and use it in the prompt instead. This way one of the lines above is chosen. If you want to have one perspective / emotion / pose each prompt, you should use `__perspective__, __emotion__, __pose__` and not curly braces or pipes.

**Comment by lgwjames** [comment]
thanks, will check

**Comment by lgwjames** [comment]
opened 10 files and opens as normal text

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by mashb1t** [reply]
but then there's no issue, right?

**Comment by lgwjames** [comment]
new prompt to try 

__jessicaL__ , __perspective__ , __emotion__,  __pose__  wearing a __Clothing_Designers__ __clothing-female__ , standing on beach in turkey, pencil eyeliner, DSLR, parted lips, pink lipstick, silver necklace with a cross on it, straw hat, Ground level (from the side) view medium Close shot (focus on chest)

no errors yet, many thanks, will update

**Comment by lgwjames** [comment]
so far ok, but what even i add, full body image, full shot, etc im only getting now head shots

__jessicaL__  ,  wearing a __Clothing_Designers__ __clothing-female__ ,  pencil eyeliner, parted lips, pink lipstick, silver necklace with a cross on it,
![2024-07-28_19-37-57_4665](https://github.com/user-attachments/assets/62467ec5-a2a3-4f48-b13e-643fd3bb8cbf)
![2024-07-28_19-10-34_3279](https://github.com/user-attachments/assets/6a8b0372-4955-4212-9d34-62dee3813a47)

![2024-07-28_18-30-46_6557](https://github.com/user-attachments/assets/0255918e-a0b3-47da-9e97-fc68d79d80b1)

![2024-07-28_17-39-45_4071](https://github.com/user-attachments/assets/b168fa15-8bab-4eeb-b6cc-da9ddde3e666)
![2024-07-28_17-24-52_5285](https://github.com/user-attachments/assets/0561fceb-940c-4a8f-bae6-603416559934)
